src/services/completions.ts

```diff
@@ -244,6 +244,9 @@ namespace ts.Completions {
         }
 
         function addStringLiteralCompletionsFromType(type: Type, result: CompletionEntry[]): void {
+            if (type && type.flags & TypeFlags.TypeParameter) {

```

Comment:
If you get the type constraint and the type is a `keyof` type, where is that handled below?